pkg/reconciler/buildrun/buildrun.go

```diff
@@ -131,6 +130,9 @@ func (r *ReconcileBuildRun) Reconcile(request reconcile.Request) (reconcile.Resu
 			// Set the LastTaskRunRef in the BuildRun status
 			buildRun.Status.LatestTaskRunRef = &generatedTaskRun.Name
 			ctxlog.Info(ctx, "updating BuildRun status with TaskRun name", namespace, request.Namespace, name, request.Name, "TaskRun", generatedTaskRun.Name)
+			serviceAccountName := &generatedTaskRun.Spec.ServiceAccountName
+			ctxlog.Info(ctx, "updating BuildRun serviceAccountName with serviceAccount retrieved from TaskRun", "BuildRun", buildRun.Name, "ServiceAccountName", serviceAccountName)
+			buildRun.Status.ServiceAccountName = serviceAccountName
```

Comment:
Please extend the comment. Also I suggest to merge the two log statements into one.